addons/viewport/src/components/Tool.tsx

```diff
@@ -0,0 +1,258 @@
+    list.find(i => i.default) ||
+    responsiveViewport;
+
+  useEffect(() => {
+    const eventProps: ViewportChangedEventProps = { item, isRotated };
+    api.emit(EVENT_VIEWPORT_CHANGED, eventProps);
+  }, [item, isRotated]);
+
+  const ref = useRef<ViewportStyles>();
+
+  const styles = getStyles(ref.current, item.styles, isRotated);
+
+  useEffect(() => {
+    ref.current = styles;
+  }, [item]);
+
+  if (disable || Object.entries(viewports).length === 0) {
+    return null;
+  }
+
+  return (
+    <Fragment>
+      <WithTooltip
+        placement="top"
+        trigger="click"
+        tooltip={({ onHide }) => (
+          <TooltipLinkList links={toLinks(list, item, setState, state, onHide)} />
+        )}
+        closeOnClick
+      >
+        <IconButtonWithLabel
+          key="viewport"
+          title="Change the size of the preview"
+          active={!!styles}
+          onDoubleClick={() => {
+            setState({ ...state, selected: responsiveViewport.id });
+          }}
+        >
+          <Icons icon="grow" />
+          {styles ? (
+            <IconButtonLabel>
+              {isRotated ? `${item.title} (L)` : `${item.title} (P)`}
+            </IconButtonLabel>
+          ) : null}
+        </IconButtonWithLabel>
+      </WithTooltip>
+
+      {styles ? (
+        <ActiveViewportSize>
+          <Global
+            styles={{
+              [`#${iframeId}`]: {
+                margin: `auto`,
+                transition: 'width .3s, height .3s',
+                position: 'relative',
+                border: `${theme.layoutMargin}px solid black`,
+                borderRadius: theme.appBorderRadius,
+                boxShadow:
+                  '0 0 100px 1000px rgba(0,0,0,0.5), 0 4px 8px 0 rgba(0,0,0,0.12), 0 2px 4px 0 rgba(0,0,0,0.08)',
+
+                ...styles,
+              },
+              [`#${wrapperId}`]: {
+                padding: theme.layoutMargin,
+                alignContent: 'center',
+                alignItems: 'center',
+                justifyContent: 'center',
+                justifyItems: 'center',
+                overflow: 'auto',
+
+                display: 'grid',
+                gridTemplateColumns: '100%',
+                gridTemplateRows: '100%',
+              },
+            }}
+          />
+          <ActiveViewportLabel title="Viewport width">
+            {styles.width.replace('px', '')}
+          </ActiveViewportLabel>
+          <IconButton
+            key="viewport-rotate"
+            title="Rotate viewport"
+            onClick={() => {
+              setState({ ...state, isRotated: !isRotated });
+            }}
+          >
+            <Icons icon="transfer" />
+          </IconButton>
+          <ActiveViewportLabel title="Viewport height">
+            {styles.height.replace('px', '')}
+          </ActiveViewportLabel>
+        </ActiveViewportSize>
+      ) : null}
+    </Fragment>
+  );
+};
+
+export const ViewportTool = withTheme(memo(ViewportToolComponent));
```

Comment:
HOCs inlining cause problem with typings. But now git thinks that I deleted and created file. Huh, I separate move and edit to avoid this but it did not help j_j.

Reply:
perhaps you could try using `useTheme` from `@storybook/theming`?

I'm not sure it's used anywhere yet, but it should be available, I think

Reply:
This remove & add makes it super hard to review right